diff -ru avrdude-5.3.1/ser_posix.c avrdude-5.3.1-serjtag/ser_posix.c
--- avrdude-5.3.1/ser_posix.c	2006-12-12 01:02:45.000000000 +0900
+++ avrdude-5.3.1-serjtag/ser_posix.c	2007-06-18 15:41:00.000000000 +0900
@@ -66,6 +66,7 @@
   { 57600,  B57600 },
   { 115200, B115200 },
   { 230400, B230400 },
+  { 3000000, B3000000 },
   { 0,      0 }                 /* Terminator. */
 };
 
diff -ru avrdude-5.3.1/ser_win32.c avrdude-5.3.1-serjtag/ser_win32.c
--- avrdude-5.3.1/ser_win32.c	2006-12-11 21:47:35.000000000 +0900
+++ avrdude-5.3.1-serjtag/ser_win32.c	2007-06-18 15:40:36.000000000 +0900
@@ -46,6 +46,7 @@
 /* HANDLE hComPort=INVALID_HANDLE_VALUE; */
 
 static struct baud_mapping baud_lookup_table [] = {
+#define CBR_3000000	3000000
   { 1200,   CBR_1200 },
   { 2400,   CBR_2400 },
   { 4800,   CBR_4800 },
@@ -54,6 +55,9 @@
   { 38400,  CBR_38400 },
   { 57600,  CBR_57600 },
   { 115200, CBR_115200 },
+#ifdef CBR_3000000
+  { 3000000, CBR_3000000 },
+#endif
   { 0,      0 }                 /* Terminator. */
 };
 
